manuscript/manuscript.Rmd

```diff
@@ -242,39 +238,84 @@ When the agent cannot fully observe the state of the system, reinforcement learn
-Conversely, if we know the state-value function, we would have to perform a maximization over an expected value to find the optimal action, $a_t^* = \underset{a_t}{\text{argmax}} \, \mathbb{E}_{s_{t+1} \sim \,T(s_{t+1} \mid s_t, a_t)}[V^\pi(s_{t+1})]$, which is less tractable.
-Yet, learning the advantage function is often preferred in practice as advantage estimators favorably have lower variance than state-value and Q-value estimators [cite schulman gae].
+  \mathcal{E} = r(s_t, a_t) + \gamma \, \underset{a}{\text{max}} \, Q_\phi(s_{t+1}, a) - Q_\phi(s_t, a_t).
+$$
+
+See the appendix/reference for details on how the Bellman Error is derived from eq. [@CITE]. 
+The general scheme for approximate value-learning with deep neural networks is to sample trajectories from the environment, evaluate the Bellman Error across these samples -- with the samples indexed by $i$ --, and then perform the following gradient descent:
+
+$$
+  \phi_{t+1} = \phi_t - \alpha \nabla_\phi \, \sum_i \mathcal{E}_i^2 
+$$
+
+When being evaluated, a deep Q-learning agent will select actions by inputting the current state to the Q network and then identifying the state-action pair with the highest estimated Q value.
+The benefit of using neural networks to approximate Q-functions is that neural networks are able to generalize Q functions over unseen state-action pairs, whereas classic methods like TD-learning are not able to generalize [@CITE].
+But there a host of problems with getting neural-network-based value-learning agents to work in practice; see @CITE for more insight on these issues.   
+The primary concern with these methods is that they tend to suffer from bias which is introduced by fitting a value network to a target that incorporates an estimated value function [@CITE].  
+
+## Actor-Critic
+Actor-critic algorithms integrate the main components from both value-learning and policy gradient based methods.
+The "actor" attempts to learn the policy, and the "critic" attempts to learn a value function.
+While the general algorithm for the critic is exactly the same as mentioned in the value-learning section, the actor differs from vanilla policy gradient algorithms by incorporating the critic in its gradient ascent step.
+Instead of estimating the gradient of the return as in eq ..., the gradient of the return is estimated as 
+
+$$
+  \nabla_\theta J(\pi_\theta) \approx \sum_{i,t} \nabla_\theta \log{\pi_\theta (a_{i,t} \mid s_{i,t})} \, \hat{Q}(s_{i,t}, a_{i, t}).
+$$
+
+where $\hat{Q}$ is a value function estimated by the critic.
+The benefit of actor-critic algorithms is that they balance the issues with bias and variance that are common with value-learning and policy gradient methods respectively [@CITE].
+Consequently, actor-critic algorithms have achieved a large majority of the state of the art results in model-free deep RL [@CITE].  
+
 
 **Survey of RL Algorithms**
```

Comment:
Now that the above section is more flushed out, I'm wondering if we can fold the "Survey" section into the above.  E.g. maybe we can reference the table of algorithms in the last subsection (on actor-critic and other methods) and then move onto examples?  (I haven't done a word-count recently but I think we're going to have issues...)